src/analytics/private/segments/segments-model.ts

```diff
@@ -0,0 +1,195 @@
+ */
+
+interface Segment {
+  segmentId: string;
+  segmentType: 'User' | 'Session' | 'Event';
+  name: string;
+  description: string;
+  projectId: string;
+  appId: string;
+  createBy: string;
+  createAt: number;
+  lastUpdateBy: string;
+  lastUpdateAt: number;
+  refreshSchedule: RefreshSchedule;
+  eventBridgeRuleArn?: string;
+  criteria: SegmentCriteria;
+}
+
+interface RefreshSchedule {
+  cron: 'Manual' | 'Daily' | 'Weekly' | 'Monthly' | 'Custom';
+  hour?: number;
+  minute?: number;
+  weeklyDay?: 'Sunday' | 'Monday' | 'Tuesday' | 'Wednesday' | 'Thursday' | 'Friday' | 'Saturday';
+  monthlyDay?: number;
+  cronExpression?: string;
+  expireAfter: number; // timestamp
+}
+
+type FilterOperator = 'and' | 'or';
+
+interface SegmentCriteria {
+  filterGroups: SegmentFilterGroup[];
+  operator: FilterOperator;
+}
+
+interface SegmentFilterGroup {
+  description?: string;
+  startDate?: string;
+  endDate?: string;
+  relativeDateRange?: string; // TODO: expression
+  filters: SegmentFilter[];
+  operator: FilterOperator;
+}
+
+type SegmentFilter = UserSegmentFilter; // Add "SessionSegmentFilter" and "EventSegmentFilter" in the future
+
+interface UserSegmentFilter {
+  conditions: UserSegmentFilterCondition[];
+  operator: FilterOperator;
+}
+
+type UserSegmentFilterCondition =
+  UserEventCondition
+  | EventsInSequenceCondition
+  | UserAttributeCondition
+  | UserInSegmentCondition;
+
+enum SegmentFilterConditionType {
+  UserEventCondition,
+  EventsInSequenceCondition,
+  UserAttributeCondition,
+  UserInSegmentCondition,
+}
+
+interface UserEventCondition {
+  conditionType: SegmentFilterConditionType.UserEventCondition;
+  hasDone: boolean;
+  event: EventWithParameter;
+  metricCondition: MetricCondition;
+}
+
+interface EventsInSequenceCondition {
+  conditionType: SegmentFilterConditionType.EventsInSequenceCondition;
+  hasDone: boolean;
+  events: EventWithParameter[];
+  isInOneSession: boolean;
+  isDirectlyFollow: boolean;
+}
+
+interface UserAttributeCondition {
+  conditionType: SegmentFilterConditionType.UserAttributeCondition;
+  attributeCondition: ParameterCondition;
+}
+
+interface UserInSegmentCondition {
+  conditionType: SegmentFilterConditionType.UserInSegmentCondition;
+  isInSegment: boolean;
+  segmentId: string;
+}
+
+interface EventWithParameter {
+  eventName: string;
+  eventParameterConditions?: ParameterCondition[];
+}
+
+interface ParameterCondition {
+  parameterType: ParameterType;
+  parameterName: string;
+  dataType: ParameterDataType;
+  conditionOperator: ConditionOperator;
+  inputValue: number | string;
+}
+
+interface MetricCondition {
+  metricType: SegmentFilterEventMetricType;
+  conditionOperator: ConditionOperator;
+  inputValue: number;
+}
+
+enum ParameterType {
+  PREDEFINED = 'predefined',
+  CUSTOM = 'custom',
+}
+
+enum ParameterDataType {
+  STRING = 'string',
+  INTEGER = 'integer',
+  DOUBLE = 'double',
+  FLOAT = 'float',
+}
+
+enum ConditionOperator {
```

Comment:
Could we reuse it with the existing SQL builder? @llmin

Reply:
SQL builder has similar enum `ExploreAnalyticsOperators`, which has values like `not_in`. I want to change it to `not in` so as to be used directly. Since the segments sql construct logic is complicated and different from that of explore, I'd rather to implement separate sql builder for segments.